smap_l2_gridder/crs.py

```diff
@@ -0,0 +1,259 @@
+    'BASEGEOGCRS["WGS 84",ENSEMBLE["World Geodetic System 1984 ensemble", '
+    'MEMBER["World Geodetic System 1984 (Transit)", ID["EPSG",1166]], '
+    'MEMBER["World Geodetic System 1984 (G730)", ID["EPSG",1152]], '
+    'MEMBER["World Geodetic System 1984 (G873)", ID["EPSG",1153]], '
+    'MEMBER["World Geodetic System 1984 (G1150)", ID["EPSG",1154]], '
+    'MEMBER["World Geodetic System 1984 (G1674)", ID["EPSG",1155]], '
+    'MEMBER["World Geodetic System 1984 (G1762)", ID["EPSG",1156]], '
+    'MEMBER["World Geodetic System 1984 (G2139)", ID["EPSG",1309]], '
+    'MEMBER["World Geodetic System 1984 (G2296)", ID["EPSG",1383]], '
+    'ELLIPSOID["WGS 84",6378137,298.257223563,LENGTHUNIT["metre",1,'
+    'ID["EPSG",9001]],ID["EPSG",7030]], ENSEMBLEACCURACY[2],'
+    'ID["EPSG",6326]],ID["EPSG",4326]],'
+    'CONVERSION["US NSIDC EASE-Grid 2.0 North",'
+    'METHOD["Lambert Azimuthal Equal Area",'
+    'ID["EPSG",9820]],PARAMETER["Latitude of natural origin",90,'
+    'ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9102]],ID["EPSG",8801]],'
+    'PARAMETER["Longitude of natural origin",0,'
+    'ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9102]],ID["EPSG",8802]],'
+    'PARAMETER["False easting",0,LENGTHUNIT["metre",1,ID["EPSG",9001]],'
+    'ID["EPSG",8806]],PARAMETER["False northing",0,LENGTHUNIT["metre",1,'
+    'ID["EPSG",9001]],ID["EPSG",8807]],ID["EPSG",6929]],CS[Cartesian,2,'
+    'ID["EPSG",4469]],AXIS["Easting (X)",South,MERIDIAN[90.0,'
+    'ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9102]]]],'
+    'AXIS["Northing (Y)",South,MERIDIAN[180.0,'
+    'ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9102]]]],'
+    'LENGTHUNIT["metre",1,ID["EPSG",9001]],ID["EPSG",6931]]'
+)
+
+GPD_TO_WKT = {
+    "EASE2_N09km.gpd": epsg_6931_wkt,
+    "EASE2_M09km.gpd": epsg_6933_wkt,
+}
+
+
+def geotransform_from_target_info(target_info: dict) -> Geotransform:
+    """Return a geotransform from the grid_info dict.
+
+    grid_info contains a parsed NSIDC gpd file.
+
+    GT(0) x-coordinate of the upper-left corner of the upper-left pixel.
+    GT(1) w-e pixel resolution / pixel width.
+    GT(2) row rotation (typically zero).
+    GT(3) y-coordinate of the upper-left corner of the upper-left pixel.
+    GT(4) column rotation (typically zero).
+    GT(5) n-s pixel resolution / pixel height (negative value for a north-up image).
+
+    """
+    validate_gpd_style(target_info)
+    return Geotransform(
+        target_info['Map Origin X'],
+        target_info['Grid Map Units per Cell'],
+        np.float64(0.0),
+        target_info['Map Origin Y'],
+        np.float64(0.0),
+        -1.0 * target_info['Grid Map Units per Cell'],
+    )
+
+
+def validate_gpd_style(target_info: dict) -> None:
+    """Raise error if the gpd is non-standard."""
+    if (target_info['Grid Map Origin Column'] != -0.5) or (
+        target_info['Grid Map Origin Row'] != -0.5
+    ):
+        raise InvalidGPDError('Can not use non standard gpd.')
```

Comment:
> This first version only works on SPL2SMP_E data so I know that it's a L2G dataset on the 9km EASE-Grid2. and I added the GPD files to this repository that represent the information for those grids so that I can parse them directly.

Ah, this answers the question I was trying to ask! Thanks